src/plugins/layout/stateprovider.ts

```diff
@@ -188,13 +260,18 @@ export function shouldIgnoreNode(node: Node, shadowDom: ShadowDom): boolean {
   // Ignore subtrees of ignored nodes (e.g. text with a <script> parent)
   if (!ignore) {
     let parentIndex = getNodeIndex(node.parentNode);
-    if (parentIndex !== null) {
-      let parentShadowNode = shadowDom.getShadowNode(parentIndex);
-      assert(!!parentShadowNode, "shouldIgnoreNode", "parentShadowNode is missing");
-      if (parentShadowNode && parentShadowNode.ignore && parentShadowNode.node !== document) {
```

Comment:
Did you lose the parentShadowNode.node !== document check intentionally ?

Reply:
No, if you look at the addition, you will notice I've:
if (parentIndex !== null && parentIndex > 0) and also a comment:
// Check if parent is not null or not document node (parentIndex === 0)